pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -156,8 +161,10 @@ func (r *InferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			}
 
 			// remove our finalizer from the list and update it.
-			isvc.ObjectMeta.Finalizers = utils.RemoveString(isvc.ObjectMeta.Finalizers, finalizerName)
-			if err := r.Update(context.Background(), isvc); err != nil {
+			controllerutil.RemoveFinalizer(isvc, finalizerName)
```

Comment:
Can we link the reference to best practice in the comment?

Reply:
I don't have a link unfortunately. I personally saw some small problems with `update`, for example, for fields like `resources.requests.memory`, if you set a value `1024Mi`, after a `update`, it will be updated to `1Gi`. Even though there's no difference, but tools like Argo CD consider there's a diff.

Things like this is a not big problem but kind of annoying. But using patch to only update the specified scope of the spec is much safer.